server/container_create.go

```diff
@@ -403,6 +403,15 @@ func (s *Server) CreateContainer(ctx context.Context, req *types.CreateContainer
 		}, nil
 	}
 
+	systemCtx, err := s.contextForNamespace(req.SandboxConfig.Metadata.Namespace)
+	if err != nil {
+		return nil, fmt.Errorf("get context for namespace: %w", err)
+	}
+
+	if err := s.StorageImageServer().IsRunningImageAllowed(ctx, &systemCtx, req.Config.Image.Image, req.Config.Image.UserSpecifiedImage); err != nil {
```

Comment:
⚠️ This is racy and insufficient.

The policy check must happen _exactly_ on the resolved image (`imageID := imgResult.ID` in `container_create_linux.go`. Making a name lookup here is racy WRT concurrent pulls changing what `Image` resolves to.

Reply:
That's a great point. Thanks. I will update the code.